jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -5673,9 +5673,43 @@ private static ReflectionFactory getReflectionFactory() {
 	/*[ENDIF] JAVA_SPEC_VERSION >= 11 */
 
 /*[IF JAVA_SPEC_VERSION >= 20]*/
-	/* ToDo: Real implementation to be added under https://github.com/eclipse-openj9/openj9/issues/15933. */
+	/**
+	 * {@return an unmodifiable set of the {@linkplain AccessFlag access
+	 * flags} for this class, possibly empty}
+	 *
+	 * <p> If the underlying class is an array class, then its
+	 * {@code PUBLIC}, {@code PRIVATE} and {@code PROTECTED}
+	 * access flags are the same as those of its component type.  If this
+	 * {@code Class} object represents a primitive type or void, the
+	 * {@code PUBLIC} access flag is present, and the
+	 * {@code PROTECTED} and {@code PRIVATE} access flags are always
+	 * absent. If this {@code Class} object represents an array class, a
+	 * primitive type or void, then the {@code FINAL} access flag is always
+	 * present and the interface access flag is always
+	 * absent. The values of its other access flags are not determined
+	 * by this specification.
+	 *
+	 * @see #getModifiers()
+	 * @jvms 4.1 The ClassFile Structure
+	 * @jvms 4.7.6 The InnerClasses Attribute
+	 * @since 20
+	 */
 	public Set<AccessFlag> accessFlags() {
-		return null;
+		int rawModifiers = getModifiersImpl();
```

Comment:
Is there a reason to use `getModifiersImpl()` rather than `getModifiers()` ? We do not want to duplicate the code of `getModifiers()`.

Reply:
OK, I see a comment is added explaining this.